modules/shapefile/src/lib/parsers/parse-dbf.js

```diff
@@ -17,11 +17,14 @@ class DBFParser {
     this.binaryReader = new BinaryChunkReader();
     this.textDecoder = new TextDecoder(encoding);
     this.state = STATE.START;
-    this.result = {};
+    this.result = {
+      data: []
+    };
   }
 
   write(arrayBuffer) {
-    this.binaryReader.write(arrayBuffer);
+    // TODO: fix Node streaming to produce ArrayBuffers, not Buffers
+    this.binaryReader.write(toArrayBuffer(arrayBuffer));
     this.state = parseState(this.state, this.result, this.binaryReader, this.textDecoder);
```

Comment:
We could just make the binaryReader do the slicing? It is already partially byteOffset aware..

Reply:
I'm not sure I understand what you're getting at. `parseState` calls `binaryReader.getDataView` internally.